types/jest/index.d.ts

```diff
@@ -724,8 +752,11 @@ declare namespace jest {
         /**
          * Ensure that an object is an instance of a class.
          * This matcher uses `instanceof` underneath.
+         *
+         * Optionally, you can provide a type for the expected value via a generic.
+         * This is particuarly useful for ensuring expected objects have the right structure.
          */
-        toBeInstanceOf(expected: any): R;
+        toBeInstanceOf<E>(expected: E): R;
```

Comment:
```suggestion
        toBeInstanceOf<E = any>(expected: E): R;
```

Explicit fallback rather than implicit